src/Randomizer.SMZ3.Tracking/AutoTracking/AutoTracker.cs

```diff
@@ -288,7 +288,7 @@ protected void Connector_MessageReceived(object? sender, EmulatorDataReceivedEve
         /// <summary>
         /// Sends requests out to the connected lua script
         /// </summary>
-        protected void SendMessages()
+        protected async void SendMessages()
```

Comment:
...but you also shouldn't use `async void` unless you have no other choice (e.g. event handlers). So SendMessages should be `async Task` and Connector_Connected should be `async void` (because the event delegate specifies a void return type).

Async void still runs synchronously (because there is no `Task` to `await`)

Reply:
Huh. Didn't realize that. I'm surprised visual studio doesn't warn you about that reading up on it since it seems like events are the only general use case async void functions and can be really bad otherwise. 🤔 But sorry, I'll keep that in mind. I guess I need to brush up on all the complexities of C# tasks as they seem to be a consistent weak point of mine.

Reply:
I'm guessing that's because there's no way to see whether a method is an event handler from the method definition alone.

And yeah, async programming takes a while to get a good understanding of. I still don't fully understand all of the possible deadlocks and race conditions but I can recognize them at least. 

In general, you want to use `async Task`, `async Task<T>` or `async ValueTask<T>` where possible. `async IAsyncEnumerable<T>` is another fun new one.